extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClient.java

```diff
@@ -185,25 +347,94 @@ private HttpUrl getHealthUrl() {
     private HttpUrl getSecretUrl(String key, String entryType) {
         key = URLEncoder.encode(key, StandardCharsets.UTF_8);
 
-        // restore '/' characters to allow sub-directories
+        // restore '/' characters to allow subdirectories
         key = key.replace("%2F", "/");
 
-        final var vaultApiPath = hashicorpVaultConfig.getVaultApiSecretPath();
+        var vaultApiPath = configValues.secretPath();
 
-        return Objects.requireNonNull(HttpUrl.parse(hashicorpVaultConfig.getVaultUrl()))
+        return getVaultUrl()
                 .newBuilder()
                 .addPathSegments(PathUtil.trimLeadingOrEndingSlash(vaultApiPath))
                 .addPathSegment(entryType)
                 .addPathSegments(key)
                 .build();
     }
 
+    @NotNull
+    private Request httpGet(HttpUrl requestUri) {
+        return new Request.Builder()
+                .url(requestUri)
+                .headers(headers)
+                .get()
+                .build();
+    }
+
+    @NotNull
+    private Request httpPost(HttpUrl requestUri, Object requestBody) {
+        return new Request.Builder()
+                .url(requestUri)
+                .headers(headers)
+                .post(createRequestBody(requestBody))
+                .build();
+    }
+
+    @NotNull
+    private Headers getHeaders() {
+        var headersBuilder = new Headers.Builder().add(VAULT_REQUEST_HEADER, Boolean.toString(true));
+        headersBuilder.add(VAULT_TOKEN_HEADER, configValues.token());
+        return headersBuilder.build();
+    }
+
+    /**
+     * Executes the specified request synchronously. Failed requests will be retried with an exponential backoff (delay)
+     * defined as {@code exponentialBackoff = base^retries}.
+     * Given an exponential base of 1.5s the backoff will be 1.5s, 2.25, 3.375s, 5.09s, 7.59s, 11.3s, ...
+     * <p>
+     * Requests will be retried if
+     * <ol>
+     *     <li>the response status code is retryable and</li>
+     *     <li>execution time (delay) is within the remaining time</li>
+     * </ol>
+     * <p>
+     * This method is not relying on httpClient.execute(request, fallbacks) method since the http client retry policy
+     * might not fit the recovery time of the Vault.
+     *
+     * @param request        the request to execute
+     * @param timeoutSeconds timeout which will break the loop ultimately
+     * @return result of the http request
+     * @throws IOException          httpclient was not able to execute the request
+     * @throws ExecutionException   result of a retried http request could not be retrieved
+     * @throws InterruptedException execution was interrupted while waiting for the result of a retried http request
+     */
+    @NotNull
+    private Response executeWithRetry(Request request, long timeoutSeconds) throws IOException, ExecutionException, InterruptedException {
```

Comment:
retry is handled automatically by the `EdcHttpClient` component, you can pass a `FallbackFactory` to tell when the call needs to be retried (take a look at `FallbackFactories` for examples), please avoid this custom implementation.

Reply:
But, I cannot influence for how long it will be retried since those configurations are set globally for every request.

Imagine a scenario where the K8S Cluster & EDC boots with a fresh token of TTL=30 Minutes.
Now Vault takes 3 Minutes to boot for whatever reason. 
Will the default http retrypolicy be sufficient enough in this use case? The way I see it I have no control over that.

Also the Vault can also go into recovery mode while running so this does not affect start up phase only.

The EDC shouldn't enter unhealthy states because the retry setting wasn't sufficient enough.

In the descrption above I proposed to move this logic into the `EdcHttpClient` since I (also) think that it's misplaced here but that would require a different issue and PR.

Reply:
I cannot see a case where EDC starts and fundamental services (like Vault or database) are not already online.
I don't think we should manage hypothetical cluster issues here. 

The logic should be: is the service down/not staisfying the connector requests for some reasons? the connector will try again for some time, if after some time the external service is not able to execute what is requested, then the issue is at the network side.

Having the retry policy being configurable for every call is a different feature that needs to be discussed and tackled in a different discussion/issue/pr